server/service/transport_osquery.go

```diff
@@ -73,10 +74,25 @@ func decodeSubmitDistributedQueryResultsRequest(ctx context.Context, r *http.Req
 		results[query] = queryResults
 	}
 
+	// Statuses were represented by strings in osquery < 3.0 and now
+	// integers in osquery > 3.0. Massage to string for compatibility with
+	// the service definition.
+	statuses := map[string]string{}
+	for query, status := range shim.Statuses {
+		switch s := status.(type) {
+		case string:
+			statuses[query] = s
+		case float64:
```

Comment:
is it really a float and not an int?

Reply:
JSON numbers always become `float64` when you unmarshal them into `interface{}`.